lib/src/priority_queue.dart

```diff
@@ -267,10 +316,10 @@ class HeapPriorityQueue<E> implements PriorityQueue<E> {
     // in the heap will also have lower priority.
     do {
       var index = position - 1;
-      var element = _queue[index]!;
+      var element = _elementAt(index);
       var comp = comparison(element, object);
-      if (comp == 0) return index;
-      if (comp < 0) {
+      if (comp <= 0) {
+        if (comp == 0 && element == object) return index;
```

Comment:
This turned out to break at least one project's tests.

I think it's the right thing to do, and they were able to work around it.

Reply:
Super. If you rely on `compareTo` for equality, and don't override `==` to match, then I guess this would affect you.